components/library/libjpeg/pkg5

```diff
@@ -1,11 +1,11 @@
 {
     "dependencies": [
-        "SUNWcs",
         "system/library"
     ],
     "fmris": [
+        "image/library/libjpeg",
```

Comment:
Is my understanding correct that `image/library/libjpeg` will contain version `9e` after this change?

The `image/library/libjpeg` package is currently not needed by any other package/component in the oi-userland.  Why you cannot create new `image/library/libjpeg-9` (or similar) package without touching rest of libjpeg packages?

Reply:
The change happened due to libjpeg-turbo being faster at the time. However we ended up bringing more maintenance burden into the system because of the small speed up. There is no need to name the package differently just because of a version change since there is no plan to keep around multiple versions for longer than we need.  Libjpeg is mostly considered finished so I expect little need for updates other than security upgrades. Neither do most linux distros.

As to the main situation, we must have a main libjepg in the default location As mentioned in the openssl case which falls into the same category some build scripts cannot be changed to build against libraries not under /usr/lib in the case of our 32bit to 64bit transition we added ISA subdirectories (i386/amd64) and there we also needed to invent hacks like runpath editing etc. to be able to package some software at all.

If it is unnecessary to put libraries into non-default paths we should not do so. This library is also required to be in the default path by all the packages in #16235 hence the name change to simply libjepg to reflect that.

Reply:
To be clear, basically nobody uses this library as default libjpeg library for different reasons, technical and non-technical.

Reply:
@mtelka Hello, not sure if you noticed the comments above. I made the switch to libjpeg-turbo back then. I can answer a few questions that you may have about the state of the userland w.r.t libjpeg.

Reply:
> @mtelka Hello, not sure if you noticed the comments above. I made the switch to libjpeg-turbo back then.

Sure, I noticed.  BTW, the change from IJG 6 to turbo was very good change.

Yes, you are saying that _others_ do not use `libjpeg9` and almost exclusively use `libjpeg-turbo`.  That means that we _probably_ want to go the same way.  I'm just asking to be sure what exactly we want to achieve because it is not rare that SunOS does many things _differently_ than others :-).

BTW, if we want to go the `libjpeg-turbo`-only way (i.e. with no IJG 6 and no IJG 9) then I see this PR as a step in wrong direction (hence my questions, see below).

> I can answer a few questions that you may have about the state of the userland w.r.t libjpeg.

I think the current state of `libjpeg` in oi-userland is clear to me, so I currently have no questions about the _current state_.  Maybe the only exception is my question re the API v6 compatibility between IJG and turbo, but the question is not oi-userland specific.  Please review what I wrote [here](https://github.com/OpenIndiana/oi-userland/pull/16035#issuecomment-1967078529) and if you find any issue, please let me know.

I do have questions about the _future_ and about stuff outside the oi-userland.  I'd appreciate at least some answers to my questions.  Either from you @alarcher, or from @Toasterson, or from @AndWac, or from ... anybody else who have anything to add to the matter :-).  The most important question is the first one:  _Do we want/need both `libjpeg-turbo` and `libjpeg 9` as the final state? If not, which one? Why that one?_

Thank you.

Reply:
Hi @alarcher thanks for this info. I would need to look at which package exactly started this change, but we started to plan the move because there were some components starting to complain about wanting libjpeg9 and refusing to compile. From the two linux distros I have at hand (Ubuntu and archlinux) both do indeed use libjpeg-turbo as their base. At least Ubuntu manages to have both libjpeg9 and libjpeg-turbo under their equivalent of `/usr/lib/amd64` which sounds to me like the most sane approach for all edge cases. With that and a properly setup pkgconfig we should be ok for whatever comes in the future. libjpeg-turbo stated that they do not want to adopt the jpeg9 API as it was too small a change to warrant the effort and Color management was deemed too exotic apparently. Such statements tend to cause software devs to abandon the library. I can set up a test system to see how complex a PR I would need to place a copy of libjpeg-turbo under `/usr/lib/amd64` but we need to get rid of libjpeg6. We lost Solaris ABI compatibility some time ago so there is no longer a justification to keep libjpeg6 around.

Reply:
You may want to add libjpeg9 for component strictly requiring it but the vast majority want the usual 6-8 ABI backward compatible library. So the answer is: having both is good but the default should remain libjpeg-turbo version 8. Another argument I had read was about DRM-related changes in IJG's version. Back then I added all the flavours because I was not sure whether some libaries required strictly libjpeg-turbo 6, 7, or 8. Then later on I moved most components (nearly all of them) to the latest version after checking that the component worked. I had left libjpeg6-ijg in place as some binary distributions from OpenSolaris times expected it in /usr/lib and I preferred to avoid breaking things unnecessarily for some people. Since I was doing this in my spare time I made decisions based on the time available mostly and have been more conservative than required. Years have passed so compatibility is not relevant anymore. Thanks for improving the situation.

Reply:
Thanks for these notes, @alarcher.

For now I'd prefer to stay with `libjpeg-turbo` only and add `libjpeg9` only after there is some software really requiring it.  @Toasterson: Do we already have something in the queue where missing `libjpeg9` would be a showstopper?

Reply:
It's been sometime that andreas and I looked at it. At the moment I can't find any in my notes. So we can leave libjpeg9 out of the picture for this PR.